torch/csrc/jit/script/compiler.cpp

```diff
@@ -696,9 +696,17 @@ struct to_ir {
   }
 
   // see [setstate type]
-  static TypePtr getTypeForSetStateArg(const Self* self) {
+  static TypePtr getTypeForSetStateArg(const Def& def, const Self* self) {
     TORCH_CHECK(self, "Expected __setstate__ to have a `self` argument");
-    self->getClassType()->getMethod("__getstate__")->ensure_defined();
```

Comment:
Was the segfault because the result of `getMethod("__getstate__")` was nullptr?

Reply:
yes